src/machine.rs

```diff
@@ -868,4 +907,82 @@ mod test {
         assert_eq!(machine.registers[Register::SP as usize], 100);
         assert_eq!(machine.registers[Register::FLAGS as usize], 0b11);
     }
+        #[test]
+    fn run_fibo() {
+        let program = rv16asm! {
+            //Inicializando os registradores
+            "MOV A, #0",
+            "MOV B, #1",
+            "MOV M, #0",
+            
+            //Indo de 0 até 9, ou seja 10 elementos
+            "GTE M, #9",
+            "CJP #40",
+ 
+            //Soma B em A, calcula o proximo numero sequencia de fibonacci
+            "ADD A, B",
+
+            //Use a stack para mover o valor de A para C
+            "SUB SP, #2",
+            "STR A, SP",
+            "LDR C, SP",
+            "ADD SP, #2",
+
+            //Use a stack para mover o valor de B para A
+            "SUB SP, #2",
+            "STR B, SP",
+            "LDR A, SP",
+            "ADD SP, #2",
+
+            //Usa a staack para mover o valor de C para B
+            "SUB SP, #2",
+            "STR C, SP",
+            "LDR B, SP",
+            "ADD SP, #2",
+
+            //Reinicia o loop
+            "ADD M, #1",
+            "JMP #6",
+
+            "ADD FLAGS, #1"
+
+        };
+
+        let mut mem = LinearMemory::new(66000);
+        assert!(mem.write_program(&program));
+
+        let mut machine = Machine::new(mem);
+        machine.set_register(Register::SP, 0x100);
+        while let Ok(_) = machine.step() {
+            //machine.print_regs();
+            //println!("")
+        }
+
+        machine.print_regs();
+        //assert_eq!(machine.registers[Register::FLAGS as usize], 0b0000) // the FLAGS should be 0...101
+    }
+           #[test]
+    fn run_expo() {
+        let program = rv16asm! {
+            "MOV A, #2",
+            "MOV B, #5",
+
+            "OR FLAGS, #4",   
+            "MUL A, B",     
+            "LDR SP, C",  
+        };
+
+        let mut mem = LinearMemory::new(66000);
+        assert!(mem.write_program(&program));
+
+        let mut machine = Machine::new(mem);
+        machine.set_register(Register::SP, 0x100);
+        while let Ok(_) = machine.step() {
+            //machine.print_regs();
+            //println!("")
+        }
+
+        machine.print_regs();
+        //assert_eq!(machine.registers[Register::FLAGS as usize], 0b0000) // the FLAGS should be 0...101
+    }
 }
```

Comment:
Few points here:
- You should fix the code formatting
- The test is asserting nothing, so you should verify if the operation is correct.
- Why using MUL instruction given that you've implemented `EXPR` instruction that performs exponential?